### DISCUSSION TITLE
Is MediaPlayer audio playback supported in winUI 3 desktop apps?

### DISCUSSION THREAD

**Discussion by brabebhin** [microsoft/WindowsAppSDK#1929] [Q&A] [answered]
Hello,

I have begun experimenting with migrating my media player from UWP to WinUI 3 desktop app, and I found the following that audio playback does not seem to work. I got video playback going fine with frame server mode, but there is no audio in sight.

Any thoughts?

`
var firstVideoFile = (await KnownFolders.VideosLibrary.GetFilesAsync()).LastOrDefault();
            if (firstVideoFile != null)
            {
                player.AutoPlay = false;
                player.AudioDevice = (await Windows.Devices.Enumeration.DeviceInformation.FindAllAsync(Windows.Devices.Enumeration.DeviceClass.AudioRender)).FirstOrDefault();
                player.AudioCategory = MediaPlayerAudioCategory.Movie;
                player.MediaEnded += Player_MediaEnded;
                player.MediaFailed += Player_MediaFailed;
                player.MediaOpened += Player_MediaOpened;
                playerElement.SetMediaPlayer(player);
                MediaSource source = MediaSource.CreateFromStorageFile(firstVideoFile);
                source.OpenOperationCompleted += Source_OpenOperationCompleted;
                player.Source = source;
                player.IsMuted = false;
                player.SystemMediaTransportControls.IsEnabled = true;
                player.SystemMediaTransportControls.IsPlayEnabled = true;
                player.SystemMediaTransportControls.IsPauseEnabled = true;
            }`

**Comment by brabebhin** [comment, marked as answer]
Never-mind, it is just me getting rusty and not checking the codecs...

**Reply by jamers99** [reply]
Are there docs for this? Not finding much with a quick Google search beyond this...
https://blogs.windows.com/windowsdeveloper/2022/01/28/adding-event-handlers-sounds-dark-mode-windowing-part-2/
